core/src/consensus/mod.rs

```diff
@@ -1122,8 +1119,6 @@ impl ConsensusGraph {
                 Ok(log) => blocks_to_skip.contains(&log.block_hash),
                 Err(_) => false,
             })
-            .skip(offset)
-            .take(limit)
```

Comment:
`config.get_logs_filter_max_limit` should be used here, otherwise we might unnecessarily process a lot of logs.

Reply:
Do you mean instead of checking get_logs_filter_max_limit in "fn get_logs(&self, filter: CfxRpcLogFilter)" and "fn logs(&self, filter: EthRpcLogFilter)", checking get_logs_filter_max_limit in this function here, and return error if over max limitation?

Reply:
Currently, this PR will process all the logs in `ConsensusGraph::logs`, and then report an error from `impl/cfx/get_logs` if we have more than `get_logs_filter_max_limit` results.

It is enough to process just `get_logs_filter_max_limit + 1` matching logs to know that we 're over the limit.

So I suggest you add something like this:

```rust
.take(self.config.get_logs_filter_max_limit + 1)
```